tests/test_worker.py

```diff
@@ -0,0 +1,857 @@
+import asyncio
+import concurrent.futures
+import logging
+import logging.handlers
+import multiprocessing
+import queue
+import time
+import uuid
+from dataclasses import dataclass
+from datetime import datetime, timedelta, timezone
+from typing import Any, Callable, Iterable, List, Optional, Tuple
+
+import pytest
+
+import temporalio.activity
+import temporalio.client
+import temporalio.exceptions
+import temporalio.worker
+from tests.helpers.worker import (
+    KSAction,
+    KSExecuteActivityAction,
+    KSWorkflowParams,
+    Worker,
+)
+
+
+async def test_activity_hello(client: temporalio.client.Client, worker: Worker):
+    async def say_hello(name: str) -> str:
+        return f"Hello, {name}!"
+
+    result = await _execute_workflow_with_activity(
+        client, worker, say_hello, "Temporal"
+    )
+    assert result.result == "Hello, Temporal!"
+
+
+async def test_activity_info(client: temporalio.client.Client, worker: Worker):
+    # Make sure info call outside of activity context fails
+    assert not temporalio.activity.in_activity()
+    with pytest.raises(RuntimeError) as err:
+        temporalio.activity.info()
+    assert str(err.value) == "Not in activity context"
+
+    # Capture the info from the activity
+    info: Optional[temporalio.activity.Info] = None
+
+    async def capture_info() -> None:
+        nonlocal info
+        info = temporalio.activity.info()
+
+    result = await _execute_workflow_with_activity(
+        client, worker, capture_info, start_to_close_timeout_ms=4000
+    )
+
+    assert info
+    assert info.activity_id
+    assert info.activity_type == "capture_info"
+    assert info.attempt == 1
+    assert info.heartbeat_details == []
+    assert info.heartbeat_timeout is None
+    # TODO(cretz): Broken?
+    # assert info.schedule_to_close_timeout is None
```

Comment:
I need to open a server issue about this. This is being set to a few nanos more than `start_to_close_timeout` by the server for some reason.